pandas/tests/io/test_clipboard.py

```diff
@@ -69,18 +72,36 @@ def teardown_class(cls):
     def check_round_trip_frame(self, data_type, excel=None, sep=None,
                                encoding=None):
         data = self.data[data_type]
-        data.to_clipboard(excel=excel, sep=sep, encoding=encoding)
-        if sep is not None:
-            result = read_clipboard(sep=sep, index_col=0, encoding=encoding)
+        if excel in [None, True] and sep is not None and len(sep) > 1:
+            with tm.assert_produces_warning():
```

Comment:
Typically for raising exceptions and warnings we split this off into separate tests